src/ast/nodes/CallExpression.ts

```diff
@@ -1,23 +1,35 @@
 import CallOptions from '../CallOptions';
+import { DeoptimizableEntity } from '../DeoptimizableEntity';
 import { ExecutionPathOptions } from '../ExecutionPathOptions';
 import {
 	EMPTY_IMMUTABLE_TRACKER,
 	ImmutableEntityPathTracker
 } from '../utils/ImmutableEntityPathTracker';
-import { EMPTY_PATH, ObjectPath, UNKNOWN_EXPRESSION, UNKNOWN_PATH } from '../values';
+import {
+	EMPTY_PATH,
+	LiteralValueOrUnknown,
+	ObjectPath,
+	UNKNOWN_EXPRESSION,
+	UNKNOWN_PATH,
+	UNKNOWN_VALUE
+} from '../values';
 import Identifier from './Identifier';
 import * as NodeType from './NodeType';
 import { ExpressionEntity } from './shared/Expression';
 import { ExpressionNode, NodeBase } from './shared/Node';
 import SpreadElement from './SpreadElement';
 
-export default class CallExpression extends NodeBase {
+export default class CallExpression extends NodeBase implements DeoptimizableEntity {
 	type: NodeType.tCallExpression;
 	callee: ExpressionNode;
 	arguments: (ExpressionNode | SpreadElement)[];
 
 	private callOptions: CallOptions;
+
+	// Caching and deoptimization:
+	// We collect deoptimization information if returnExpression !== UNKNOWN_EXPRESSION
 	private returnExpression: ExpressionEntity | null;
+	private expressionsToBeDeoptimized: DeoptimizableEntity[];
```

Comment:
I did not check but I would not assume it to work as well as the ImmutableEntityPathTracker in that it would deoptimize more literal values than necessary e.g. in situations where one origin needs more than one lookup to get its value. At least these were the reason why the immutable tracker was used here.